src/diagnostics.rs

```diff
@@ -215,7 +215,7 @@ pub fn report_error<'tcx, 'mir>(
     report_msg(
         *ecx.tcx,
         DiagLevel::Error,
-        &if let Some(title) = title { format!("{}: {}", title, msg) } else { msg.clone() },
```

Comment:
If the goal is to avoid redundancy, we should also do something about the `title == None` case here.

The only errors that can trigger this are `MultipleSymbolDefinitions { .. } | SymbolShimClashing { .. }`... so maybe we should just come up with a title for them? Any suggestions?